docs/developer-docs/security/security-best-practices/dos.mdx

```diff
@@ -4,32 +4,52 @@ keywords: [security, concept, denial of service, dos, cycles, captcha]
 
 import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
 
-# Security best practices: Denial of service
+# Security Best Practices: Denial of Service
 
 <MarkdownChipRow labels={["Intermediate", "Security", "Concept"]} />
 
-## Protect against draining the cycles balance
 
-### Security concern
+## Protect Against DoS and DDoS Attacks
 
-Canisters pay for their cycles which makes them inherently vulnerable to attacks that consume all their cycles.
+### Security Concern
+
+A Denial of Service (DoS) attack aims to make a system unavailable by overwhelming it with requests or data. A Distributed Denial of Service (DDoS) attack is a more sophisticated version, where the attack originates from multiple sources, making it harder to block. An attacker will typically search for operations that are free to be executed by anyone but which are expensive for the application in terms of certain resources such as storage, memory usage, network bandwidth, computing resources, etc. In the case of canisters, such attacks can aim to deplete cycles, making the canister unable to process legitimate requests. The reverse gas model means that a dapp needs to implement strategies to deal with this.
 
 ### Recommendation
 
-* Consider monitoring, early authentication, and rate limiting on the canister level to mitigate this. Also, be aware that an attacker will aim for the call consuming most cycles. See the "Cycle balance drain attacks section" in [how to audit an ICP canister](https://www.joachim-breitner.de/blog/788-How_to_audit_an_Internet_Computer_canister).
+To protect your canisters from DoS and DDoS attacks, consider the following strategies:
+* **Bot prevention techniques**: Use methods like captchas or proof of work to ensure only legitimate users can access your canister. Captchas help verify that the user is human, while proof of work requires the user to spend computational resources to proceed, deterring automated attacks. An example implementation can be found in the [Internet Identity project](https://github.com/dfinity/internet-identity).
```

Comment:
Does II also implement PoW or only Captcha?
Could you link the piece of code that does this?